framework/wazuh/core/cdb_list.py

```diff
@@ -128,48 +129,156 @@ def split_key_value_with_quotes(line, file_path='/CDB_LISTS_PATH'):
             # Check that the line finishes with ..."
             # Check that the line has the structure ...:"...
             if second_quote != len(line) - 1 or line[first_quote - 1: first_quote + 1] != ':"':
-                raise WazuhError(1800, extra_message={'path': join('WAZUH_HOME', file_path)})
+                raise WazuhError(1800, extra_message={'path': path.join('WAZUH_HOME', file_path)})
 
     # There is an odd number of quotes (or more than 4)
     else:
-        raise WazuhError(1800, extra_message={'path': join('WAZUH_HOME', file_path)})
+        raise WazuhError(1800, extra_message={'path': path.join('WAZUH_HOME', file_path)})
 
     return key, value
 
 
-def get_list_from_file(path):
-    """Get CDB list from file
+def get_list_from_file(path, raw=False):
+    """Get CDB list from a file.
 
-    :param path: Relative path of list file to get
-    :return: CDB list
+    Parameters
+    ----------
+    path : string
+        Full path of list file to get.
+    raw : bool, optional
+        Respond in raw format.
+
+    Returns
+    -------
+    result : dict, str
+        CDB list.
     """
-    file_path = join(common.ossec_path, path)
-    output = list()
+    result = dict()
 
     try:
-        with open(file_path) as f:
-            for line in f.read().splitlines():
+        with open(path) as f:
+            output = f.read()
+
+        if raw:
+            result = output
+        else:
+            for line in output.splitlines():
                 if 'TEMPLATE' not in line:
```

Comment:
We talked about checking TEMPLATE only as the key of the first line?

Reply:
Replaced by a regex that will skip empty lines or line which start (not contain) `TEMPLATE:`

```PYTHON
def get_list_from_file(path, raw=False):
    """..."""
    # Match empty lines or lines which start with TEMPLATE:
    regex_template = re.compile(r'^TEMPLATE:|^\s*$')
    result = dict()

    try:
        with open(path) as f:
            output = f.read()

        if raw:
            result = output
        else:
            for line in output.splitlines():
                if not re.match(regex_template, line):
                    if '"' not in line:
                        # Check if key and value are not surrounded by double quotes
                        key, value = line.split(':')
                    else:
                        # Check if key and/or value are surrounded by double quotes
                        key, value = split_key_value_with_quotes(line, path)
                    result.update({key: value})
```